unime/src/routes/prompt/credential-offer/+page.svelte

```diff
@@ -17,74 +11,22 @@
   import TopNavBar from '$src/lib/components/molecules/navigation/TopNavBar.svelte';
   import { state } from '$src/stores';
 
-  import Check from '~icons/ph/check-bold';
   import DownloadSimple from '~icons/ph/download-simple-fill';
-  import PlugsConnected from '~icons/ph/plugs-connected-fill';
-  import Question from '~icons/ph/question';
-  import WarningCircle from '~icons/ph/warning-circle-fill';
-  import X from '~icons/ph/x-bold';
-
-  // const {
-  //   elements: { root, input },
-  //   helpers: { isChecked }
-  // } = createCheckbox({});
-
-  // if ($state.current_user_prompt == null) {
-  //   goto('/scan');
-  // }
-
-  console.log($state.current_user_prompt);
-  // TODO: type
-  let credential_offer: any = $state.current_user_prompt?.credential_offer ?? {};
-  console.log({ credential_offer });
-
-  let issuer_name = $state.current_user_prompt?.issuer_name;
-
-  // const checkboxes = credential_offer.credentials.map((c, i) => {
-  //   const {
-  //     elements: { root, input },
-  //     helpers: { isChecked }
-  //   } = createCheckbox();
-  //   return { c, root, input, isChecked };
-  // });
 
-  // console.log(checkboxes);
-
-  // const map = credential_offer.credentials.map((credential, i) => {
-  //   const {
-  //     elements: { root, input },
-  //     helpers: { isChecked }
-  //   } = createCheckbox();
-  //   // return { credential, root, input, isChecked };
-  //   return { credential };
-  // });
-
-  // console.log({ map });
-
-  let all_offer_indices = credential_offer.credentials.map((_, i) => i);
-
-  let issuerLogoUrl: string | null;
-  // let credentialLogoUrl: string | null; // TODO: batch credentials, see "all_offer_indices"
-
-  // let images = new Map<string, string>();
+  // TODO: generate binding in core
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  interface CredentialOffer {
+    credential_issuer: string;
+    credentials: any[];
+    grants: any;
+  }
+  /* eslint-enable @typescript-eslint/no-explicit-any */
 
-  let credentialLogoUrls = {};
+  let credential_offer: CredentialOffer = $state.current_user_prompt?.credential_offer;
 
-  onMount(async () => {
-    issuerLogoUrl = await getImageAsset('client_0', true);
-    // credentialLogoUrl = await getImageAsset('credential', true);
+  let issuer_name: string = $state.current_user_prompt?.issuer_name;
 
-    // $state?.current_user_prompt?.display?.forEach(async (display, i) => {
-    //   if (display.logo_uri) {
-    //     credentialLogoUrl = await getImageAsset('credential', true, display.logo_uri);
-    //   }
-    // });
-    $state.current_user_prompt.display.forEach(async (display, i) => {
-      if (display.logo?.url) {
-        credentialLogoUrls[i] = await getImageAsset(`credential_${i}`, true);
-      }
-    });
-  });
+  let all_offer_indices: string[] = credential_offer.credentials.map((_, i: number) => i.toString());
```

Comment:
`all_offer_indices` should be a `number[]` because otherwise this error will occur:
```
04-09 12:07:38.374 15423 15533 I         : [2024-04-09][10:07:38][INFO][webview::unknown] Dispatching action: {"type":"[Credential Offer] Selected","payload":{"offer_indices":["0","1","2","3","4"]}}
04-09 12:07:38.386 15423 15423 I Tauri/Console: File: http://tauri.localhost/node_modules/.vite/deps/@tauri-apps_plugin-log.js?v=8cb7f683 - Line 111 - Msg: [2024-04-09][10:07:38][INFO][webview::unknown] Dispatching action: {"type":"[Credential Offer] Selected","payload":{"offer_indices":["0","1","2","3","4"]}}
04-09 12:07:38.391 15423 15533 E         : [2024-04-09][10:07:38][ERROR][webview::unknown] invalid args `action` for command `handle_action`: invalid type: string "0", expected usize
```

Perhaps you can also delete the `ts_rs` attribute in this Action:
```rust
#[derive(Serialize, Deserialize, Debug, TS, Clone)]
#[ts(export, export_to = "bindings/actions/CredentialOffersSelected.ts")]
pub struct CredentialOffersSelected {
    #[ts(type = "Array<string>")]     // <-- remove this line
    pub offer_indices: Vec<usize>,
}
```

And update its binding.

Reply:
Thanks for catching that!